From a26abb07884834fc72778565fdae8fddae3948fa Mon Sep 17 00:00:00 2001
From: Patrick McCarty <patrick.mccarty@intel.com>
Date: Wed, 11 Jul 2018 17:20:47 -0700
Subject: [PATCH 2/2] Disable man page install when not generated

Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 doc/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/doc/CMakeLists.txt b/doc/CMakeLists.txt
index ffdb721..077f9ef 100644
--- a/doc/CMakeLists.txt
+++ b/doc/CMakeLists.txt
@@ -11,6 +11,7 @@ ADD_CUSTOM_TARGET (doc-man
 ADD_CUSTOM_TARGET (doc)
 ADD_DEPENDENCIES (doc doc-html doc-man)
 
+if (NOT WITH_MAN EQUAL 0)
 INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/dnf.plugin.builddep.8
     ${CMAKE_CURRENT_BINARY_DIR}/dnf.plugin.config_manager.8
     ${CMAKE_CURRENT_BINARY_DIR}/dnf.plugin.copr.8
@@ -34,3 +35,4 @@ if (${WITHOUT_LOCAL} STREQUAL "0")
 INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/dnf.plugin.local.8
 	DESTINATION share/man/man8)
 endif()
+endif()
-- 
2.18.0

